src/Framework/TestSuite.php

```diff
@@ -497,22 +499,18 @@ public function addTestFiles(iterable $fileNames): void
 
     /**
      * Counts the number of test cases that will be run by this test.
+     *
+     * @todo refactor usage of numTests in DefaultResultPrinter
      */
-    public function count(bool $preferCache = false): int
+    public function count(): int
     {
-        if ($preferCache && $this->cachedNumTests !== null) {
-            return $this->cachedNumTests;
-        }
-
-        $numTests = 0;
+        $this->numTests = 0;
 
         foreach ($this as $test) {
-            $numTests += \count($test);
+            $this->numTests += \count($test);
```

Comment:
What do you think about making this explicit?

```suggestion
            $this->numTests += $test->count();
```

Reply:
I will refactor the counting mechanism(s?) for the data provider changes :)